src/NetworkViewUi.lua

```diff
@@ -163,45 +181,125 @@ function M.update_items(player_index)
-        tooltip = { "", item_name, ": ", item.count }
+        def.tooltip = item_tooltip(item.item, item.count)
+        if is_item then
+          def.tags = { event = UiConstants.NV_ITEM_SPRITE, item = item.item }
+        end
       else
-        item_name = game.fluid_prototypes[item.item].localised_name
-        tooltip = {
-          "",
-          item_name,
-          ": ",
-          string.format("%.0f", item.count),
-          " at ",
-          { "format-degrees-c", string.format("%.0f", item.temp) },
-        }
+        def.tooltip = fluid_tooltip(item.item, item.temp, item.count)
       end
-      local item_view = item_h_stack.add({
-        type = "sprite-button",
-        elem_type = net_view.view_type,
-        sprite = sprite_path .. "/" .. item.item,
-        tooltip = tooltip,
-      })
+      local item_view = item_h_stack.add(def)
       item_view.number = item.count
     end
+    pad_item_row(item_h_stack, 10 - #row)
+  end
+  if not did_pad then
+    pad_item_row(item_flow.add(h_stack_def), 10)
+  end
+end
+
+function M.on_gui_click_item(event, element)
+  --[[
+  This handles a click on an item sprite in the item view.
+  If the cursor has something in it, then the cursor content is dumped into the item network.
+  If the cursor is empty then we grab something from the item network.
+    left-click grabs one item.
+    shift + left-click grabs one stack.
+    ctrl + left-click grabs it all.
+  ]]
+  local player = game.players[event.player_index]
+  if player == nil then
+    return
+  end
+  local inv = player.get_main_inventory()
+
+  -- if we have an empty cursor, then we are taking items, which requires a valid target
+  if player.is_cursor_empty() then
+    local item_name = event.element.tags.item
+    if item_name == nil then
+      return
+    end
+
+    local network_count = GlobalState.get_item_count(item_name)
+    local stack_size = game.item_prototypes[item_name].stack_size
+
+    if event.button == defines.mouse_button_type.left then
+      -- shift moves a stack, non-shift moves 1 item
+      local n_transfer = 1
+      if event.shift then
+        n_transfer = stack_size
+      elseif event.control then
+        n_transfer = network_count
+      end
+      -- move one item or stack to player inventory
+      n_transfer = math.min(network_count, n_transfer)
+      if n_transfer > 0 then
+        local n_moved = inv.insert({name = item_name, count = n_transfer})
+        if n_moved > 0 then
+          GlobalState.set_item_count(item_name, network_count - n_moved)
+          element.number = GlobalState.get_item_count(item_name)
+          element.tooltip = item_tooltip(item_name, element.number)
+        end
+      end
+    end
+    return
+
+  else
+    -- There is a stack in the cursor. Deposit it.
+    local cs = player.cursor_stack
+    if not cs or not cs.valid_for_read then
+      return
+    end
+
+    -- don't deposit tracked entities (can be unique)
+    if cs.item_number ~= nil then
```

Comment:
Is this check for unique entities like spidertrons or armor? Nice catch. Currently network chests don't check for this so I'm thinking if this should be added there.

Reply:
Yeah, I was thinking what would happen if I put my decked-out Spidertron in the item-network. All gone.
I don't know how to fix that, short of implementing a true inventory, so I just added protection here.
I'd be OK with preventing the addition of anything in a non-clean state. I just didn't know how.  I suppose we could check for a custom name and a non-empty inventory (grid/trunk/logistic/modules/ammo/etc).
We could also dump all inventories when adding a complex item to the item network.

Reply:
For now I'd say it's not a huge priority. I haven't had a single bug ticket around this behavior so I'm fine leaving it as-is and waiting for someone to report it as an issue.

Worst case scenario, people can always just re-load the last autosave.